server/src/main/java/org/apache/iotdb/db/qp/executor/PlanExecutor.java

```diff
@@ -228,9 +229,12 @@ private QueryDataSet processDataQuery(QueryPlan queryPlan, QueryContext context)
     if (queryPlan instanceof AlignByDevicePlan) {
       queryDataSet = new AlignByDeviceDataSet((AlignByDevicePlan) queryPlan, context, queryRouter);
     } else {
-      if (queryPlan instanceof GroupByPlan) {
+      if (queryPlan.getPaths() == null || queryPlan.getPaths().isEmpty()) {
+        // no time series are selected, return EmptyDataSet
+        return new EmptyDataSet();
+      } else if (queryPlan instanceof GroupByPlan) {
         GroupByPlan groupByPlan = (GroupByPlan) queryPlan;
-        return queryRouter.groupBy(groupByPlan, context);
+        queryDataSet =  queryRouter.groupBy(groupByPlan, context);
```

Comment:
No need to do this change, the limit and offset are ignored in group by.

Reply:
Or you can add a JIRA to report this, actually in syntax, we support limit and offset in group by, but we just ignore these two in the code.

Reply:
I updated the docs